dbt/adapters/snowflake/impl.py

```diff
@@ -158,5 +159,31 @@ def quote_seed_column(self, column: str, quote_config: Optional[bool]) -> str:
         else:
             return column
 
+    @available
+    def standardize_grants_dict(self, grants_table: agate.Table) -> dict:
+        grants_dict = {}
+
+        for row in grants_table:
+            grantee = row['grantee_name']
+            privilege = row['privilege']
+            granted_by = row['granted_by']
+
+            # super gross, but it works
+            # (this is the value from profiles.yml)
+            current_role = self.connections.profile.credentials.role
```

Comment:
TODO:
- MUST: I've remembered that `role` is technically not required in `profiles.yml`. If not specified, Snowflake will opt for the user's configured `default_role` when they connect. So, if this value is missing, we may need to run `select current_role` after all. I think our best bet would be to do that once at the start of the invocation, rather than doing it over and over.
- SHOULD: This should really be a new adapter method, rather than reaching deep in as we're doing here. That method would also allow us to abstract over whether `credentials.role` is provided, or needs to be accessed by introspective database query.

Reply:
Another option (!!): https://docs.snowflake.com/en/sql-reference/functions/result_scan.html

This would let us do it all in SQL — though a bit weird, to be honest, since we'd need to pass it a specific `query_id` after running the first query. We can get that `query_id`, though. Worth a shot...?

To be honest, if this proves too tricky to get this working, we can revert to the previous (simpler) logic that just filtered out any grants of the `OWNERSHIP` privilege, and treated all the rest as fair game.

Reply:
@dbeatty10 Interested in your gut reaction between those options. (No need for full investigation.) It does feel like the simplest thing here is reverting to your + Matt's previous logic, which just filters out the `OWNERSHIP` privilege.

Reply:
will revert, and push up...DONE